crates/db_schema/src/lib.rs

```diff
@@ -24,3 +24,37 @@ pub mod source;
 pub mod traits;
 #[cfg(feature = "full")]
 pub mod utils;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(EnumString, Display, Debug, Serialize, Deserialize, Clone, Copy)]
+pub enum SortType {
+  Active,
+  Hot,
+  New,
+  TopDay,
+  TopWeek,
+  TopMonth,
+  TopYear,
+  TopAll,
+  MostComments,
+  NewComments,
+}
+
+#[derive(EnumString, Display, Debug, Serialize, Deserialize, Clone, Copy, PartialEq)]
+pub enum ListingType {
+  All,
+  Local,
+  Subscribed,
+  Community,
+}
+
+#[derive(EnumString, Display, Debug, Serialize, Deserialize, Clone, Copy)]
+pub enum SearchType {
+  All,
+  Comments,
+  Posts,
+  Communities,
+  Users,
+  Url,
+}
```

Comment:
These enums are very useful for api clients, and should be available without the `full` feature, so i had to move them.